src/decoders/http2/http2_decoder.c

```diff
@@ -338,6 +346,28 @@ bool get_request_headers(qd_http2_decoder_t *decoder)
         decoder->frame_length_processed += 1;
     }
 
+    //
+    // Before the call to decompress_headers(), we need to make sure that there is some data left in the scratch buffer before we decompress it.
+    //
+    int buffer_data_size = scratch_buffer->size - scratch_buffer->offset;
+    int contains_pad_length = scratch_buffer->size - pad_length;
+    int pad_length_offset = scratch_buffer->size - pad_length - scratch_buffer->offset;
+    bool valid_pad_length = contains_pad_length > 0;
+    bool valid_buffer_data = buffer_data_size > 0;
+    bool valid_pad_length_offset = pad_length_offset > 0;
+    if(decoder->frame_payload_length == 0 || !valid_buffer_data || !valid_pad_length || !valid_pad_length_offset) {
```

Comment:
'if' is not a function call.  Please put spaces after ifs (both styles are found throughout this file).

Reply:
Replaced `if(` with `if (`